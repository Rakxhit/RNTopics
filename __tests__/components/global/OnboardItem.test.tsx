import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import OnboardItem from '../../../src/components/global/OnboardItem';
jest.mock('react-native-linear-gradient', () => 'LinearGradient');

describe('Onboarding item', () => {
  const mockOnPressFirst = jest.fn();
  const mockOnPressSecond = jest.fn();
  const imageSource = {uri: 'https://gif.png'};
  const title = 'Test title';
  const subTitle = 'Test Subtitle';
  const buttonTitleFirst = 'First Button';
  const buttonTitleSecond = 'Second Button';

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const {toJSON} = render(
      <OnboardItem
        buttonTitleFirst={buttonTitleFirst}
        imageSource={imageSource}
        onPressFirst={mockOnPressFirst}
        subtitle={subTitle}
        title={title}
        buttonTitleSecond={buttonTitleSecond}
        onPressSecond={mockOnPressSecond}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correctly with one button', () => {
    const {getByText, getByTestId} = render(
      <OnboardItem
        title={title}
        subtitle={subTitle}
        buttonTitleFirst={buttonTitleFirst}
        imageSource={imageSource}
        onPressFirst={mockOnPressFirst}
      />,
    );
    expect(getByText(title)).toBeTruthy();
    expect(getByText(subTitle)).toBeTruthy();
    expect(getByTestId('background-image')).toBeTruthy();
    expect(getByTestId('first-button-alone')).toBeTruthy();
    const button = getByTestId('first-button-alone');
    fireEvent.press(button);
    expect(mockOnPressFirst).toHaveBeenCalled();
  });

  it('should render correctly with the second button', () => {
    const {getByText, getByTestId} = render(
      <OnboardItem
        buttonTitleFirst={buttonTitleFirst}
        title={title}
        subtitle={subTitle}
        buttonTitleSecond={buttonTitleSecond}
        imageSource={imageSource}
        onPressFirst={mockOnPressFirst}
        onPressSecond={mockOnPressSecond}
      />,
    );
    expect(getByText(title)).toBeTruthy();
    expect(getByText(subTitle)).toBeTruthy();
    expect(getByTestId('background-image')).toBeTruthy();
    expect(getByText(buttonTitleSecond)).toBeTruthy();
    expect(getByText(buttonTitleFirst)).toBeTruthy();
    const firstButton = getByTestId('first-button');
    fireEvent.press(firstButton);
    expect(mockOnPressFirst).toHaveBeenCalled();
    const secondButton = getByTestId('second-button');
    fireEvent.press(secondButton);
    expect(mockOnPressSecond).toHaveBeenCalled();
  });
});
