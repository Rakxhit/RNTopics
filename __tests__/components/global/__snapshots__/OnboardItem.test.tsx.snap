// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding item should match the snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "flex-end",
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={
      {
        "uri": "https://gif.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="background-image"
  />
  <LinearGradient
    colors={
      [
        "rgba(0,0,0,0.0)",
        "rgba(0,0,0,0.2)",
        "rgba(0,0,0,0.8)",
        "rgba(0,0,0,1)",
      ]
    }
    style={
      {
        "flex": 1,
        "justifyContent": "flex-end",
        "padding": 20,
      }
    }
  >
    <View
      style={
        {
          "marginBottom": 40,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Poppins-SemiBold",
            "fontSize": 26,
            "marginBottom": 10,
          }
        }
      >
        Test title
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontFamily": "Poppins-Regular",
            "fontSize": 12,
            "marginBottom": 30,
          }
        }
      >
        Test Subtitle
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#11b9a8",
              "borderRadius": 30,
              "opacity": 1,
              "paddingVertical": 15,
              "width": "45%",
            }
          }
          testID="first-button"
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 15,
              }
            }
          >
            First Button
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 30,
              "opacity": 1,
              "paddingVertical": 15,
              "width": "45%",
            }
          }
          testID="second-button"
        >
          <Text
            style={
              {
                "color": "#0D1829",
                "fontFamily": "Poppins-SemiBold",
                "fontSize": 15,
              }
            }
          >
            Second Button
          </Text>
        </View>
      </View>
    </View>
  </LinearGradient>
</View>
`;
